### PR TITLE
match longest shortcut by default

### PR DESCRIPTION
**Pull request by taooceros**
Previous behavior:
- Shortcut will be replaced in order they are created, so if two shortcut share same prefix the early one will be replaced even if the later one is longer.

Correct behavior:
- Match longest shortcut first

This only appears when using `@{key}` as this will match prefix.

### CODE REVIEW

**Comment by jjw24** [comment]
Sorry why is matching the longest one the correct behaviour?

**Comment by taooceros** [comment]
> Sorry why is matching the longest one the correct behaviour?

well assume you have two shortcut `sp` `spotify` for and `sps` for `spotify search`, I would assume if user type `@sps some song` that they mean `spotify search some song` rather than `spotifys some song`.

And this would align with how the whole string replace works.

**Comment by jjw24** [comment]
Why is there a replace logic if shortcut key does not match? If shortcut is set as `sp` for `spotify` and user types `@sps some song`, as `sps` does not match `sp`, shouldn't it not expand?

**Comment by taooceros** [comment]
> Why is there a replace logic if shortcut key does not match? If shortcut is set as `sp` for `spotify` and user types `@sps some song`, as `sps` does not match `sp`, shouldn't it not expand? Just not understanding the use case for replace part of the logic.

Well it would be hard to examine whether the variable is separated by space and do replace with that. Further there might be use case when we just want to replace part of a string for easier typing which I think is reasonable.

**Review by VictoriousRaptor — APPROVED** [review]

File: Flow.Launcher/ViewModel/MainViewModel.cs [hunk 1]
```diff
@@ -893,7 +899,7 @@ private Query ConstructQuery(string queryText, IEnumerable<CustomShortcutModel>
             StringBuilder queryBuilder = new(queryText);
             StringBuilder queryBuilderTmp = new(queryText);
 
-            foreach (var shortcut in customShortcuts)
+            foreach (var shortcut in customShortcuts.OrderByDescending(x => x.Key.Length))
```
[the remark this answers is not in the document]
Comment (taooceros) [reply Flow.Launcher/ViewModel/MainViewModel.cs:902]: Yes sorry for the format

**Comment by taooceros** [comment]
@jjw24 said we should add a comment there.

File: Flow.Launcher/ViewModel/MainViewModel.cs (hunk 1 above)
Comment (jjw24) [inline Flow.Launcher/ViewModel/MainViewModel.cs:902]:
```suggestion
            # Sorting order is important here, the reason is for matching longest shortcut by default
            foreach (var shortcut in customShortcuts.OrderByDescending(x => x.Key.Length))
```

**Review by jjw24 — APPROVED** [review]

**Pull request merged**